extensions/guacamole-auth-sso/modules/guacamole-auth-sso-base/src/main/java/org/apache/guacamole/auth/sso/SSOAuthenticationProvider.java

```diff
@@ -86,20 +93,32 @@ public SSOAuthenticationProvider(
      *     The SSOAuthenticationProviderService implementation that should be
      *     used for core authentication functions.
      *
+     * @param ssoResource
+     *     The SSOResource that should be used to manually redirect the user to
+     *     the IdP, as well as to provide any implementation-specific REST
+     *     endpoints.
+     *
      * @param modules
      *     Any additional modules that should be used when creating the Guice
      *     injector.
      */
     public SSOAuthenticationProvider(
             Class<? extends SSOAuthenticationProviderService> authService,
+            Class<? extends SSOResource> ssoResource,
             Iterable<? extends Module> modules) {
         injector = Guice.createInjector(Iterables.concat(Collections.singletonList(new AbstractModule() {
 
             @Override
             protected void configure() {
+
                 bind(AuthenticationProvider.class).toInstance(SSOAuthenticationProvider.this);
                 bind(Environment.class).toInstance(LocalEnvironment.getInstance());
                 bind(SSOAuthenticationProviderService.class).to(authService);
+
+                LinkedBindingBuilder<SSOResource> resourceBinding = bind(SSOResource.class);
```

Comment:
It's not very obvious why this extra complication is here; a comment might be nice.

Reply:
Fair. Added via rebase.